accepted/00081-image_management.md

```diff
@@ -0,0 +1,233 @@
+   Build Service.
+   Add the revision to image pillar and adjust saltboot and image-sync formulas
+   to handle it.
+   The actual image file will be created before `revision`,
+   so the file name will still contain the build_id in order to make it unique.
+
+
+`/srv/susemanager/pillar_data/images/org1/image-POS_Image_Graphical7.x86_64-7.0.0-build33.sls`
+```
+
+images:
+    POS_Image_Graphical7:  # <name>
+        7.0.0-1:           # <version>-<revision>
+            arch: x86_64
+            basename: POS_Image_Graphical7.x86_64-7.0.0
+            boot_image: POS_Image_Graphical7-7.0.0
+    ...
+            sync:
+    ...
+                bundle_url: https://server.tf.local/os-images/1//POS_Image_Graphical7.x86_64-7.0.0-build33.tar.xz
+
+```
+
+
+3. Do not re-use `suseImageInfo` entries, create new entry for each build.
+
+   For Docker the old entry can be deleted when the new image is built 
+   successfully.
+   
+   For Kiwi the entries can be deleted only manually. Typically it can
+   be deleted when the new image is synced to all Branch servers.
+
+4. Add `pillar` and `file` to `suseImageInfo`.
+
+   `pillar` points to pillar file or to the `category` column in `suseSaltPillars` table
+   if https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00080-jsonb-pillar-db-storage.md is implemented.
+   
+   `file` is a local path for images in SUMA Store or URL for
+   external images.
+   
+   Pillars and local image files will be deleted when the image is deleted 
+   in UI or API. 
+   
+   `file` will be used to show image URL in UI.
+
+5. Add `suseDeltaImageInfo` table with columns
+   - `sourceImageID`
+   - `targetImageID`
+   - `pillar`
+   - `file`
+   
+   Add API to create/delete entries, use it in `retail_create_delta`.
+
+   Display deltas in Image details UI.
+
+   Deleting source or target image would delete also the delta image.
+
+6. External stores
+
+    Make sure that API allows moving the image from SUMA to external store,
+    with adjusting pillars etc.
+
+    Implement a tool that moves the image from SUMA to external server or CDN.
+
+7. API
+
+   Add the new fields to Image.listImages and Image.getDetails.
+
+   Add methods to get/set image pillars.
+
+   Add a method to change `file` in SUMA store to external url.
+
+   Add a method to import Kiwi image, all the needed info will be passed 
+   as parameters, image file must exist in SUMA store before this call
+   or it can be an URL to external store.
+
+   Eventually rename Image.importImage to Image.importDockerImage
+
+8. Pillars for non-PXE images (optional)
+
+   It will be possible to add pillars for any image via API.
+   We could add a flag to image profile to control pillar generation 
+   during build.
+
+9. Pillar editor (optional)
+
+   Add UI to edit image pillars.
+
+
+## Image building workflow:
+
+1. Start building
+   - create new `suseImageInfo` entry
+   - for Kiwi: set `name`, `version`, `revision` to temporary values
```

Comment:
Please reword, this confused me wrt step 2. e.g.

```suggestion
   - for Kiwi: use temporary values for `name`, `version`, `revision`
```

Still not clear why `build_id` cannot be used.